### PR TITLE
Channel reset button in nsqadmin

### PR DESCRIPTION
**Pull request by DoraALin**
channel reset time pick & button in topic's channel page, nsqadmin

### CODE REVIEW

**Comment by DoraALin** [comment]
@absolute8511 for your review

File: nsqadmin/static/js/views/channel.hbs [hunk 1]
```diff
@@ -48,8 +48,8 @@
         {{/if}}
     </div>
     <div class="col-md-2" title="reset consume offset to specified data time, if specified timestamp is earlier(later) than timestamp of first(last) available message in queue, offset will be set to the beginning(tail) of message queue.">
-            <button class="btn btn-medium btn-primary" data-action="reset">Reset Channel Offset</button>
-            <input id="resetChannelDatetime" type="number" name="resetTime">
+        <button class="btn btn-medium btn-primary resetbtn" data-action="reset" disabled=true>Reset Channel Offset</button>
+        <input id="resetChannelDatetime" type="text" name="resetTime" maxlength="10" size="30" placeholder="input timestamp in second" pattern="[0-9]*"/>
```
Comment (absolute8511) [inline nsqadmin/static/js/views/channel.hbs]: Maybe restrict the min length to 10 too

File: consistence/nsqd_coordinator.go [hunk 2]
```diff
@@ -1223,6 +1224,14 @@ func (self *NsqdCoordinator) SearchLogByMsgTimestamp(topic string, part int, ts_
 		searchTs:         ts_sec * 1000 * 1000 * 1000,
 		ext:              tcData.topicInfo.Ext,
 	}
+
+	if comp.searchTs < int64(0) && ts_sec > int64(0) {
```
Comment (absolute8511) [inline consistence/nsqd_coordinator.go]: Maybe return error if timestamp is less than 0

File: nsqdserver/context.go [hunk 3]
```diff
@@ -260,6 +261,7 @@ func (c *context) EmptyChannelDelayedQueue(ch *nsqd.Channel) error {
 }
 
 func (c *context) SetChannelOffset(ch *nsqd.Channel, startFrom *ConsumeOffset, force bool) (int64, int64, error) {
+	fmt.Println(startFrom)
```
Comment (absolute8511) [inline nsqdserver/context.go]: remove this debug code

File: consistence/nsqd_coordinator.go [hunk 4]
```diff
@@ -1223,6 +1223,12 @@ func (self *NsqdCoordinator) SearchLogByMsgTimestamp(topic string, part int, ts_
 		searchTs:         ts_sec * 1000 * 1000 * 1000,
 		ext:              tcData.topicInfo.Ext,
 	}
+
+	if comp.searchTs < int64(0) {
+		return nil, 0, 0, fmt.Errorf("Invalid timestamp %v", ts_sec)
+	}
+
+	fmt.Println(comp)
```
Comment (absolute8511) [inline consistence/nsqd_coordinator.go]: remove this println pls

**Review by absolute8511 — APPROVED** [review]
LGTM

**Comment by absolute8511** [comment]
#38

**Pull request merged**